src/Centreon/Domain/Repository/RepositoryException.php

```diff
@@ -27,4 +27,79 @@
  */
 class RepositoryException extends \Exception
 {
+    /**
+     * Failure on authentication token retrieval
+     * @param string $centralServerAddress
+     * @return self
+     */
+    public static function failToGetToken(string $centralServerAddress): self
+    {
+        return new self(
+            sprintf(
+                _("Failed to get the auth token from Central : '%s''"),
+                $centralServerAddress
+            )
+        );
+    }
+
+    /**
+     * Failure returned from the API calling the Central
+     * @param string $details
+     * @return self
+     */
+    public static function apiRequestOnCentralException(string $details): self
+    {
+        return new self(_("Request to the Central's API failed") . (' : ') . $details);
+    }
+
+    /**
+     * Transport exception related to the client
+     * @param string $details
+     * @return self
+     */
+    public static function apiClientException(string $details): self
+    {
+        return new self(_("API calling the Central returned a Client exception") . (' : ') . $details);
+    }
+
+    /**
+     * Transport exception related to the redirection
+     * @param string $details
+     * @return self
+     */
+    public static function apiRedirectionException(string $details): self
+    {
+        return new self(_("API calling the Central returned a Redirection exception") . (' : ') . $details);
+    }
+
+    /**
+     * Transport exception related to the server
+     * @param string $message concatenated message with the central response
+     * @param string $details
+     * @return self
+     */
+    public static function apiServerException(string $message, string $details): self
+    {
+        return new self($message . (' : ') . $details);
+    }
+
+    /**
+     * Central response decoding failure
+     * @param string $details
+     * @return self
+     */
+    public static function apiDecodingException(string $details): self
+    {
+        return new self(_("Unable to convert Central's API response") . (' : ') . $details);
```

Comment:
Unable to convert Central's API response -> Unable to decode Central's API response